pkg/provider/identity.go

```diff
@@ -0,0 +1,111 @@
+package provider
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"strings"
+	"regexp"
+
+	azaci "github.com/Azure/azure-sdk-for-go/services/containerinstance/mgmt/2021-10-01/containerinstance"
+	client2 "github.com/virtual-kubelet/azure-aci/pkg/client"
+	"github.com/virtual-kubelet/virtual-kubelet/log"
+	armcontainerservice "github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/containerservice/armcontainerservice"
+	armmsi "github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/msi/armmsi"
+	azidentity "github.com/Azure/azure-sdk-for-go/sdk/azidentity"
+)
+
+func SetContainerGroupIdentity(identity *armmsi.Identity, identityType azaci.ResourceIdentityType, containerGroup *client2.ContainerGroupWrapper) {
+	if identity == nil || identityType != azaci.ResourceIdentityTypeUserAssigned {
+		return
+	}
+
+	cgIdentity := azaci.ContainerGroupIdentity{
+		Type: identityType,
+		UserAssignedIdentities: map[string]*azaci.ContainerGroupIdentityUserAssignedIdentitiesValue{
+			*identity.ID: &azaci.ContainerGroupIdentityUserAssignedIdentitiesValue{
+				PrincipalID: identity.Properties.PrincipalID,
+				ClientID: identity.Properties.ClientID,
+			},
+		},
+	}
+	containerGroup.Identity = &cgIdentity
+}
+
+func GetAgentPoolKubeletIdentity(ctx context.Context, providerResourceGroup string, providerVnetSubscriptionID string) (*armmsi.Identity, error) {
```

Comment:
You can have this function as a method for `ACIProvider` instead of passing the subID / RG:

```suggestion
func (p *ACIProvider) GetAgentPoolKubeletIdentity(ctx context.Context) (*armmsi.Identity, error) {
```